TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/drivetrain/Drivetrain.java

```diff
@@ -93,5 +98,81 @@ public static void drive(Vector drive, double r) {
+        double highestPower = 1;
+        final double max = Math.max(Math.abs(lfPower),
+                Math.max(Math.abs(lbPower), Math.max(Math.abs(rfPower), Math.abs(rbPower))));
+        if (max > 1)
+            highestPower = max;
+
+        lfPower = lfPower/highestPower;
+        rfPower = rfPower/highestPower;
+        lbPower = lbPower/highestPower;
+        rbPower = rbPower/highestPower;
 
-}
\ No newline at end of file
+        if (Math.abs(lfPower)-Math.abs(motors[0].getPower()) < DrivetrainConstants.wheelErrorRange) {
+            motorsWithProblems[0] = motorsWithProblems[0] + 1;
+            time.reset();
+        }
+        if (Math.abs(rfPower)-Math.abs(motors[1].getPower()) < DrivetrainConstants.wheelErrorRange) {
+            motorsWithProblems[1] = motorsWithProblems[1] + 1;
+            time.reset();
+        }
+        if (Math.abs(lbPower)-Math.abs(motors[2].getPower()) < DrivetrainConstants.wheelErrorRange) {
+            motorsWithProblems[2] = motorsWithProblems[2] + 1;
+            time.reset();
+        }
+        if (Math.abs(rbPower)-Math.abs(motors[3].getPower()) < DrivetrainConstants.wheelErrorRange) {
+            motorsWithProblems[3] = motorsWithProblems[3] + 1;
+            time.reset();
+        }
+
+        if (timeWithInactiveMotor >= DrivetrainConstants.requiredTimeWithInactiveMotor){
+            if (motorWithProblem == "rf" || motorWithProblem == "lb"){
+                lfPower = drive.y + r;
+                rbPower = drive.y - r;
+
+                motors[0].setPower((lfPower));
+                motors[1].setPower((0));
+                motors[2].setPower((0));
+                motors[3].setPower((rbPower));
+            }
+            else {
+
+                rfPower = drive.y - r;
+                lbPower = drive.y + r;
+                motors[0].setPower((0));
+                motors[1].setPower((rfPower));
+                motors[2].setPower((lbPower));
+                motors[3].setPower((0));
+            }
+        }
+        else {
+            if (time.milliseconds() < DrivetrainConstants.deltaTimeOfInactiveMotor)
+                timeWithInactiveMotor++;
+            else
+                timeWithInactiveMotor = 0;
+            if (motorsWithProblems[0] > motorsWithProblems[1] && motorsWithProblems[0] > motorsWithProblems[2] &&
+                    motorsWithProblems[0] > motorsWithProblems[3])
+                motorWithProblem = "lf";
+            else if (motorsWithProblems[1] > motorsWithProblems[0] && motorsWithProblems[1] > motorsWithProblems[2] &&
+                    motorsWithProblems[1] > motorsWithProblems[3])
+                motorWithProblem = "rf";
+            else if (motorsWithProblems[2] > motorsWithProblems[0] && motorsWithProblems[2] > motorsWithProblems[1] &&
+                    motorsWithProblems[2] > motorsWithProblems[3])
+                motorWithProblem = "lb";
+            else
+                motorWithProblem = "rb";
```

Comment:
again, for loop, and btw you already know which motor is with the problem (118-134)